### PR TITLE
Add tests and travis CI

### PR DESCRIPTION
**Pull request by zhengbli**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Place your settings in this file to overwrite default and user settings.
+{
+	"editor.insertSpaces": true
```
Comment (DanielRosenwasser) [inline .vscode/settings.json]: Ironically, this uses tabs. :smile: 

Change to 4 spaces I guess.

**Comment by DanielRosenwasser** [comment]
We shouldn't be committing the .d.ts files - it's usually better practice to commit the `tsd.json`.

**Comment by zhengbli** [comment]
@DanielRosenwasser any other comments?

File: tests/build.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                }
+            }
+        }
+    }
+
+    return outputLines.join('\n');
+}
+
+for (var file of fs.readdirSync('cases')) {
+    var text = fs.readFileSync('cases/' + file, 'utf8');
```
Comment (DanielRosenwasser) [inline tests/build.ts]: `const`

File: tests/build.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    let outputFilename = file.slice(0, file.lastIndexOf('.'));
```
Comment (DanielRosenwasser) [inline tests/build.ts]: outputFileName

File: tests/build.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    outputLines.push('[' + (parseInt(i) + 1) + ', ' + (markerLocation + 1) + ']: ' + token.scopes.join(' ') + ' ');
+                }
+            }
+        }
+    }
+
+    return outputLines.join('\n');
+}
+
+for (var file of fs.readdirSync('cases')) {
```
Comment (DanielRosenwasser) [inline tests/build.ts]: `fileName`

File: tests/build.ts (hunk 2 above)
Comment (DanielRosenwasser) [inline tests/build.ts]: `path.join`

File: tests/build.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    var text = fs.readFileSync('cases/' + file, 'utf8');
+    let outputFilename = file.slice(0, file.lastIndexOf('.'));
+    let grammer = file.slice(file.indexOf('.tsx')) === '.tsx' ? tsReactGrammar : tsGrammar;
```
Comment (DanielRosenwasser) [inline tests/build.ts]: `grammar`

Comment (DanielRosenwasser) [inline tests/build.ts]: Use `path.extname`

File: tests/build.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    if (!fs.existsSync('./generated')){
+        fs.mkdirSync('generated');
+    }
+    fs.writeFile('generated/' + outputFilename + '.txt', getScopesAtMarkers(text, grammer), "utf8");
```
Comment (DanielRosenwasser) [inline tests/build.ts]: `path.join('./generated', outputFileName + '.txt')

Comment (DanielRosenwasser) [inline tests/build.ts]: Just make the above suggestion `outputFilePath`

File: tests/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+	"compilerOptions": {
+		"target": "ES5",
+		"module": "commonjs"
+	},
+	"exclude": [
+		"cases",
+		"baselines",
+		"generated",
+		"node_modules"
+	]
```
Comment (DanielRosenwasser) [inline tests/tsconfig.json]: Use spaces in this file.

File: tests/test.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+	for (var file of fs.readdirSync('generated')) {
+		var generatedText = fs.readFileSync('generated/' + file, 'utf8');
+		var baselinesText = fs.readFileSync('baselines/' + file, 'utf8');
+		describe('Comparing ' + file, () => {
+			it('should be the same', () => {
+				chai.expect(compareText(generatedText, baselinesText)).to.true;
+			})
+		})
+	}
+})
```
Comment (DanielRosenwasser) [inline tests/test.ts]: Use spaces in this file

File: tests/test.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+/// <reference path="typings/mocha/mocha.d.ts" />
+/// <reference path="typings/chai/chai.d.ts" />
```
Comment (DanielRosenwasser) [inline tests/test.ts]: Instead of using `/// <reference path=... />`s, include these into the `tsconfig.json`.

File: tests/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
+{
+	"compilerOptions": {
+		"target": "ES5",
+		"module": "commonjs"
```
Comment (DanielRosenwasser) [inline tests/tsconfig.json]: Add `--noImplicitAny` to this.

**Pull request merged**